src/social/blogs.rs

```diff
@@ -702,6 +738,65 @@ decl_module! {
       Self::deposit_event(RawEvent::CommentReactionCreated(owner.clone(), comment_id, reaction_id));
     }
 
+    pub fn create_profile(origin, username: Vec<u8>, ipfs_hash: Vec<u8>) {
+      let owner = ensure_signed(origin)?;
+
+      let mut social_account = Self::get_or_new_social_account(owner.clone());
+      ensure!(social_account.profile.is_none(), MSG_PROFILE_ALREADY_EXISTS);
+      Self::is_username_valid(username.clone())?;
+      ensure!(ipfs_hash.len() == Self::ipfs_hash_len() as usize, MSG_IPFS_IS_INCORRECT);
+
+      social_account.profile = Some(
+        Profile {
+          created: Self::new_change(owner.clone()),
+          updated: None,
+          username: username.clone(),
+          ipfs_hash
+        }
+      );
+      <AccountByProfileUsername<T>>::insert(username.clone(), owner.clone());
+      <SocialAccountById<T>>::insert(owner.clone(), social_account.clone());
+      Self::deposit_event(RawEvent::ProfileCreated(owner.clone()));
+    }
+
+    pub fn update_profile(origin, update: ProfileUpdate) {
+      let owner = ensure_signed(origin)?;
+
+      let has_updates =
+        update.username.is_some() ||
+        update.ipfs_hash.is_some();
```

Comment:
Add an empty line before the next `ensure!` - it will be easier to read the code.